src/rtc.c

```diff
@@ -710,6 +733,108 @@ void RTC_Alarm_IRQHandler(void)
   HAL_RTC_AlarmIRQHandler(&RtcHandle);
 }
 
+/**
+  * @brief Attach Seconds interrupt callback.
+  * @note  stm32F1 has a second interrupt capability
+  *        other MCUs map this on their WakeUp feature
+  * @param func: pointer to the callback
+  * @retval None
+  */
+void attachSecondsIrqCallback(voidCallbackPtr func)
+{
+  RTCSecondsIrqCallback = func;
+
+#if defined(STM32F1xx)
+  HAL_RTCEx_SetSecond_IT(&RtcHandle);
+  __HAL_RTC_SECOND_CLEAR_FLAG(&RtcHandle, RTC_FLAG_SEC);
+#elif defined(STM32F030x4) || defined(STM32F030x6) || defined(STM32F030x8) || defined(STM32F070x6)
+#error "RTC has no wakeup feature for One-Second interrupt"
+#else
+  /* for MCUs using the wakeup feature : irq each second */
+#if defined(RTC_WUTR_WUTOCLR)
+  HAL_RTCEx_SetWakeUpTimer_IT(&RtcHandle, 0, RTC_WAKEUPCLOCK_CK_SPRE_16BITS, 0);
+#else
+  HAL_RTCEx_SetWakeUpTimer_IT(&RtcHandle, 0, RTC_WAKEUPCLOCK_CK_SPRE_16BITS);
+#endif /* RTC_WUTR_WUTOCLR */
+#endif /* STM32F1xx */
+
+  /* enable the IRQ that will trig the one-second interrupt */
+  HAL_NVIC_EnableIRQ(ONESECOND_IRQn);
+}
+
+/**
+  * @brief Detach Seconds interrupt callback.
+  * @param None
+  * @retval None
+  */
+void detachSecondsIrqCallback(void)
+{
+#if defined(STM32F1xx)
+  HAL_RTCEx_DeactivateSecond(&RtcHandle);
+#elif defined(STM32F030x4) || defined(STM32F030x6) || defined(STM32F030x8) || defined(STM32F070x6)
+#error "RTC has no wakeup feature for One-Second interrupt"
```

Comment:
No!
Adding this here break build for all those mcu. Only the dedicated example should add this.
See the build log of the CI!